internal/pkg/agent/cmd/logs.go

```diff
@@ -201,13 +206,40 @@ func logsCmd(streams *cli.IOStreams, cmd *cobra.Command) error {
 		modifier = addColorModifier
 	}
 
-	logsDir := filepath.Join(paths.Home(), logger.DefaultLogDirectory)
 	// uncomment for debugging
 	// fmt.Fprintf(streams.Err, "logs dir: %q", logsDir)
 
-	err := printLogs(cmd.Context(), streams.Out, logsDir, lines, follow, filter, modifier)
-	if err != nil {
-		return fmt.Errorf("failed to get logs: %w", err)
+	errChan := make(chan error)
+
+	go func() {
+		err := printLogs(cmd.Context(), streams.Out, logsDir, lines, follow, filter, modifier)
+		if err != nil {
+			errChan <- fmt.Errorf("failed to get logs: %w", err)
+		}
+		errChan <- nil
```

Comment:
You are sending on the channel twice if there is an error. But below you are only ever reading from the channel once. You either need to add an else or a return in the if statement.